### PR TITLE
small speed up in std help

### PR DESCRIPTION
**Pull request by fdncred**
# Description
Rewrote a small bit of the help code to try and increase the speed. It's not much of a help though.

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Comment by fdncred** [comment]
pinging you @amtoine to look at this when you get time. i could take or leave the help.nu changes (except for the one line that sorts) but the rust stuff I'd like to keep and land.

**Comment by amtoine** [comment]
just looking at the times first :yum: 

1. your measurements show a ~16% improvement (`(1 - (492ms + 384µs + 515ns) / (587ms + 364µs + 795ns)) * 100`)
2. on my machine, i've used the following little #8969-inspired module
```bash
# bench.nu
def "from ns" [] {
    [$in "ns"] | str join | into duration
}

export def main [
    code: closure
    -n: int = 50
    --verbose (-v): bool
    --pretty: bool
] {
    let times = (
        seq 1 $n
        | each {|i|
            if $verbose { print -n $"($i) / ($n)\r" }
            timeit { do $code } | into int | into decimal
        }
    )

    if $verbose { print $"($n) / ($n)" }

    let report = {
        mean: ($times | math avg | from ns)
        std: ($times | math stddev | from ns)
        times: ($times | each { from ns })
    }

    if $pretty {
        $"($report.mean) +/- ($report.std)"
    } else {
        $report
    }
}
```
but then
```bash
def measure [revision: string] {
    git checkout $revision

    cargo run --release -- --no-config-file -c "
        use bench.nu
        bench { std help ansi } -n 50
    | to nuon" | lines | last 1 | get 0 | from nuon
}

let before = (measure 393f424f1)
let after = (measure 4b8a63495)

print ((1 - $after.mean / $before.mean) * 100)
```
gives me `-1.6762786600016621` :thinking: 

**TL;DR: appears i see a slightly negative impact on the performance with this method** :open_mouth:

3. and on the other hand, whend doing the same manually from the REPL, i get `-18.9008037339625`, even worse :thinking:

**Comment by amtoine** [comment]
393f424f1 is the base of this branch, 4b8a63495 is the tip of the branch and *1 - a / b < 0* is equivalent to *a > b*, so having a negative result above means "after" is slower than "before" :confused:

**Comment by amtoine** [comment]
i also notice it's around 2x faster from `cargo run --release -- --no-config-file` :open_mouth:

could the config slow down all this :sweat_smile:

**Comment by fdncred** [comment]
> could the config slow down all this 😅

Yes, the amount of data in the config and other files sourced, definitely slow down things.

**Comment by fdncred** [comment]
doesn't look like this is going anywhere.
